### PR TITLE
CAMEL-16177: multicast parallel processing and encrypted stream cache

### PR DESCRIPTION
**Pull request by forsthofer**
- [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/converter/stream/CipherPair.java [hunk 1]
```diff
@@ -61,7 +60,18 @@ public Cipher getEncryptor() {
...
+     */
     public Cipher getDecryptor() {
-        return deccipher;
+        try {
+            Cipher deccipher = Cipher.getInstance(transformation);
+            deccipher.init(Cipher.DECRYPT_MODE, key, ivp == null ? null : new IvParameterSpec(ivp));
+            return deccipher;
+        } catch (GeneralSecurityException e) {
+            // should not happen
+            throw new IllegalStateException("Could not instanciate decryptor, e");
```
Comment (oscerd) [inline camel-core/src/main/java/org/apache/camel/converter/stream/CipherPair.java:74]: It should be

throw new IllegalStateException("Could not instanciate decryptor", e);

**Review by oscerd — CHANGES_REQUESTED** [review]
There is only a little problem

File: camel-core/src/main/java/org/apache/camel/converter/stream/CipherPair.java (hunk 1 above)
Comment (forsthofer) [reply camel-core/src/main/java/org/apache/camel/converter/stream/CipherPair.java]: I have corrected this in a separate commit

File: camel-core/src/test/java/org/apache/camel/processor/MultiCastParallelAndStreamCachingWithEncryptionTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/*
```
Comment (davsclaus) [inline camel-core/src/test/java/org/apache/camel/processor/MultiCastParallelAndStreamCachingWithEncryptionTest.java:1]: I think in Camel 2.x the license header may have 2 x stars in the beginning.

Can you run checkstyle

mvn compile -P sourcecheck

And verify

Comment (forsthofer) [reply camel-core/src/test/java/org/apache/camel/processor/MultiCastParallelAndStreamCachingWithEncryptionTest.java]: I added the extra star

**Pull request merged**